src/strategies/property.sml

```diff
@@ -6,45 +6,91 @@ import "strategies.property_importance";
 
 signature PROPERTY =
 sig
+    exception ParseError;
+
+    type pKind;
```

Comment:
I think this could be better as either `type kind` (so that scoped use is `Property.kind`) or `type propertyKind` (so that unscoped use is clear). Personally I would be happy with the former, and it is even shorter.